examples/01_plotting/plot_3d_map_to_surface_projection.py

```diff
@@ -27,11 +27,23 @@
 fsaverage = datasets.fetch_surf_fsaverage()
 
 ##############################################################################
-# Sample the 3D data around each node of the mesh
-# -----------------------------------------------
+# Use mesh curvature to display useful anatomical information
+# on inflated meshes
+#
+# Here, we load the curvature map of the hemisphere under study,
+# and define a surface map whose value for a given vertex is
+# 0.75 if the curvature is positive, 0.25 if the curvature is negative.
 
+import numpy as np
 from nilearn import surface
 
+curv_right = surface.load_surf_data(fsaverage.curv_right)
+curv_right_sign = (np.sign(curv_right) + 1) / 4 + 0.25
```

Comment:
That's a fair point. Don't you think the text explains what happens in a clear fashion though?

```
# Here, we load the curvature map of the hemisphere under study,
# and define a surface map whose value for a given vertex is
# 0.75 if the curvature is positive, 0.25 if the curvature is negative.
```

Reply:
Moreover, I thought it was important to give an example of how to actually use curvature maps to get the best plots.

Reply:
but we are trying to show how easy it is to make a surface plot, and if the example starts by computing its custom background map with a formula that looks a bit arbitrary, that hurts that point. we are already adding 2 arguments in the call below -- and unlike `title` or `colorbar`, the meaning of `bg_map_rescale` and `bg_on_data` cannot be guessed without reading at the documentation. so the example is getting more complex and the plotting function appears less user-friendly. I would either add a `bg_curvature_sign` to fsaverage, or move the plot with the custom background to the end of the example, in a dedicated section.

Reply:
If curvature sign is to become the preferred background (and I think it should, it looks better) I don't like the fact that we will need to pass `bg_map_rescale=False` to all surface plotting calls. Ideally the plots should look good with the default arguments. maybe we could have a `bg_map_rescale="auto"` and decide based on the number of values in the map? my guess is that for continuous maps (sulcal depth), rescaling is often ok, and for binarized maps (curv sign), rescaling results in too high contrast if the user wanted a gray background. that would work at least for sulc depth & curv sign, which should cover the majority of cases, and in the remaining cases an explicit value can be passed for `bg_map_rescale`

Reply:
> the "auto" could also take into account whether the bg map has values outside of [0, 1]

Ohhh yes, that's what I was just thinking as well!
You are right, setting "auto" everywhere with this behaviour would work, given that "auto":
- normalises background maps if values outside of [0, 1] are detected
- doesn't change the bg map otherwise

This would change maps generated using background maps whose values are in [0, 1[, but I think this is a rare usecase (I haven't encountered anyone doing this).
Then, I think I could just drop the convoluted tweaks  in this example, and in a later PR, we could ship pre-computed maps which render nice out-of-the-box with `bg_map_rescale="auto"`.

Reply:
I just committed and pushed this new behaviour, and simplified the example. Here are the first to figures of the example now, which I think looks ok for out-of-the-box maps:

![output](https://user-images.githubusercontent.com/13835654/207466015-bbac0d48-57f8-4832-b0b3-b45a9ef5c14e.png)

![newplot](https://user-images.githubusercontent.com/13835654/207466026-d85eeea5-2522-4217-8c7b-932cb28ae738.png)